public/app/plugins/panel/geomap/layers/data/MarkersLegend.tsx

```diff
@@ -1,44 +1,49 @@
 import React, { PureComponent } from 'react';
 import { stylesFactory } from '@grafana/ui';
-import { FieldType, formattedValueToString, GrafanaTheme, PanelData, ThresholdsConfig } from '@grafana/data';
+import { formattedValueToString, getFieldColorModeForField, GrafanaTheme } from '@grafana/data';
 import { css } from '@emotion/css';
 import { config } from 'app/core/config';
 import tinycolor from 'tinycolor2';
+import { DimensionSupplier } from '../../dims/types';
 
-interface Props {
-  txt: string;
-  data?: PanelData;
+export interface MarkersLegendProps {
+  color?: DimensionSupplier<string>;
+  size?: DimensionSupplier<number>;
 }
 
-interface State {}
-
-export class SimpleLegend extends PureComponent<Props, State> {
+export class MarkersLegend extends PureComponent<MarkersLegendProps> {
   style = getStyles(config.theme);
 
-  constructor(props: Props) {
+  constructor(props: MarkersLegendProps) {
     super(props);
     this.state = {};
   }
 
   render() {
-    let fmt = (v: any) => `${v}`;
-    let thresholds: ThresholdsConfig | undefined;
-    const series = this.props.data?.series;
-    if (series) {
-      for (const frame of series) {
-        for (const field of frame.fields) {
-          if (field.type === FieldType.number && field.config.thresholds) {
-            thresholds = field.config.thresholds;
-            fmt = (v: any) => `${formattedValueToString(field.display!(v))}`;
-            break;
-          }
-        }
-      }
+    const { color } = this.props;
+    console.log( 'MarkersLegend: FIELD color', color?.field );
+
+    if (!color) {
+      return <></>;
+    }
+
+    if (!color.field && color.fixed) {
+      return <div className={this.style.infoWrap}>Fixed: {color.fixed}</div>;
+    }
+
+    const colorMode = getFieldColorModeForField(color.field!);
+    if (colorMode.isContinuous && colorMode.getColors) {
+      return <div className={this.style.infoWrap}>Color scale</div>;
```

Comment:
@lean257 -- can you look at implementing this variation?  I was unable to get things to update predictably :frowning_face: it looks like you need to save and full refresh to get here